src/charm_helpers.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm helpers."""
+
+import os
+from urllib.parse import urlparse
+
+from literals import (
+    AIRBYTE_API_PORT,
+    AIRBYTE_VERSION,
+    CONNECTOR_BUILDER_SERVER_API_PORT,
+    INTERNAL_API_PORT,
+    WORKLOAD_API_PORT,
+)
+from structured_config import StorageType
+
+
+def create_env(model_name, app_name, config, state):
+    db_conn = state.database_connection
+
+    host = db_conn["host"]
+    port = db_conn["port"]
+    db_name = db_conn["dbname"]
+    db_url = f"jdbc:postgresql://{host}:{port}/{db_name}"
+
+    if state.minio:
+        minio_endpoint = construct_svc_endpoint(
+            state.minio["service"],
+            state.minio["namespace"],
+            state.minio["port"],
+            state.minio["secure"],
+        )
+
+    minio_endpoint = "http://minio:9000"
+    env = {
```

Comment:
thought: I wonder if it would be possible to have some documentation into why those particular values were selected.

For example, why those ports for `TEMPORAL_WORKER_PORTS`?

Reply:
Some of these are the default values taken from the Helm chart deployment. I can add a comment linking to it.